STYLE.md

```diff
@@ -0,0 +1,228 @@
+# Introduction
```

Comment:
It seems like there are two parallel topics being discussed here: one concerns the style guide itself, and the other pertains to how we should handle the existing pattern in the codebase.

First of all, thank you for emphasizing readability - I think many of the suggestions are good, and having a document like this will help promote consistency as well.

Regarding the second point: has there been any discussion about what to do with the current pattern? Some of the recommendations here are quite different from the approach used so far (as pointed out in other comments). Are we looking to apply these suggestions only to new code, or is there an intention to refactor existing code as well (which does carry some risk)?